### DISCUSSION TITLE
Did I understand CytoTrace kernel correct?

### DISCUSSION THREAD

**Discussion by dawe** [theislab/cellrank#767] [General]
Hi all, I am looking at how CytoTrace kernel is implemented. Is it correct to assert that the `ct_score` is the cell-wise mean of the expression of the top 200 genes most correlate with the (cell-wise) total number of expressed genes?
If so, I can't understand what is the rationale behind the idea. 
Also, would it be useful to allow parametrization of the top_n correlated genes?

**Comment by dawe** [comment]
Nevermind, I have thoroughly read the paper to get the rationale. Maybe it would be a good idea to state, somewhere in cellrank's documentation, that this method is particularly suited for developmental processes, and it won't work properly if there is not appreciable decrease of the number of expressed genes in the trajectory.

**Reply by Marius1311** [reply]
Thanks @dawe for bringing this up. I believe we have quite prominent references to the CytoTRACE paper in our docs, so that it's easy for ppl. to read up on the background. Let me know if you think otherwise. Re the comment on dev. data - I don't think all paper examples from the CytoTRACE paper are development, are they?

**Reply by Marius1311** [reply]
e.g. I recall there were some hematopoiesis examples in there

**Reply by Marius1311** [reply]
@dawe , I agree we could add a comment to our docs [here](https://cellrank.readthedocs.io/en/stable/api/cellrank.tl.kernels.CytoTRACEKernel.html#): "CytoTRACE relies on the assumption that differentiated cells express on average more genes than naive cells".  Would you mind opening a PR for this?

**Reply by dawe** [reply]
Hello @Marius1311, I have read the paper (not before reading the code, shame on me!). In the hematopoiesis example they still start from a stem cell and also the luminal/basal transition is quite extreme. I will open a PR as you suggest. If you don't mind I'd like to add a `top_n` option (default = 200). I'm considering the option to reverse the correlations order to take into account other kind of experiments in which the priming cell have an opposite phenotype. In this sense I'm thinking to scCUT&TAG experiments for close chromatin (e.g. H3K9me3 or H3K27me3).

**Reply by Marius1311** [reply]
Absolutely, please add PR for this and we'll be happy to review it!!

**Reply by Marius1311** [reply]
Sorry, I'm a bit out of the loop, @michalk8 already resolved this in https://github.com/theislab/cellrank/pull/761.

**Reply by Marius1311** [reply]
Can you explain what you mean by "reversing the correlations"? Basically, the CytoTRACE kernel just calls the PseudotimeKernel under the hood, pretending that the CytoTRACE score is a pseudotime. The PseudotimeKernel is very general, it accepts **any** measure of cell state together with a KNN graph and computes a directed transition matrix from it. If there's an option to get such a measure from cell state, i.e. something that quantifies cell plasticity on a continious scale, from scCUT&TAG experiments, this would be super cool and I would be open to intergration in CellRank.

**Reply by dawe** [reply]
@Marius1311 ctk sorts correlations of gene expression with number of expressed genes from top to bottom and takes the top 200 mostly correlated (right?). I’m thinking to the possibility of taking the top N with most negative correlation.
BTW in our lab we are working on using cellrank on epigenome data, it would be great if any of the developers could chime in
